app/Service/HistoricalHumidityProcessingService.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace App\Service;
+
+use App\Models\HistoricalHumidityProcessingReportsModel;
+
+class HistoricalHumidityProcessingService
+{
+
+    public function saveHumidityProcessToDb(): void
+    {
+        foreach (config('city_date.city_date')['year'] as $year) {
+            foreach (config('city_date.city_date')['month'] as $month) {
+                foreach (config('city_date.city_date')['city'] as $city) {
+                    if (intval($year) == 2023 && intval($month) >= 02) {
```

Comment:
while this is a "working" solution, this way you would have to push new code changes every month until the day the unvicerse collapses into a black hole...

I would advise that you only store a startDate in your  config, e.g.: 2013-01-01 00:00:00

Then instead of using foreach statement for the dates, maybe calculate runtime the current date (e.g.: today it's 2023-02) and then iterate between startDate and today.
simple example:
`
$startDate = new DateTime(config('start_date'));
$today = new DateTime();
$cities = config('cities');

while ($startDate < $today) {
  foreach ($cities as $city) {
    HistoricalHumidityProcessingReportsModel::firstOrCreate(...)
  }
  $startDate = $startDate->add(new DateInterval('P1M'));
}
`

Reply:
suggested reading:
https://www.phptutorial.net/php-oop/php-datetime/
https://www.php.net/manual/en/class.datetime.php
https://www.php.net/manual/en/class.dateinterval.php